pkg/localstorev2/internal/reserve/items.go

```diff
@@ -0,0 +1,264 @@
+}
+
+func batchBinToString(bin uint8, batchID []byte) string {
+	return path.Join(string(bin), string(batchID))
+}
+
+func (b *batchRadiusItem) String() string {
+	return path.Join(b.Namespace(), b.ID())
+}
+
+func (b *batchRadiusItem) Clone() storage.Item {
+	if b == nil {
+		return nil
+	}
+	return &batchRadiusItem{
+		Bin:     b.Bin,
+		BatchID: copyBytes(b.BatchID),
+		Address: b.Address.Clone(),
+		BinID:   b.BinID,
+	}
+}
+
+const batchRadiusItemSize = 1 + swarm.HashSize + swarm.HashSize + 8
+
+func (b *batchRadiusItem) Marshal() ([]byte, error) {
+
+	if b.Address.IsZero() {
+		return nil, errMarshalInvalidAddress
+	}
+
+	buf := make([]byte, batchRadiusItemSize)
+
+	i := 0
+
+	buf[i] = b.Bin
+	i += 1
+
+	copy(buf[i:i+swarm.HashSize], b.BatchID)
+	i += swarm.HashSize
+
+	copy(buf[i:i+swarm.HashSize], b.Address.Bytes())
+	i += swarm.HashSize
+
+	binary.BigEndian.PutUint64(buf[i:i+8], b.BinID)
+
+	return buf, nil
+}
+
+func (b *batchRadiusItem) Unmarshal(buf []byte) error {
+
+	if len(buf) != batchRadiusItemSize {
+		return errUnmarshalInvalidSize
+	}
+
+	i := 0
+	b.Bin = buf[i]
+	i += 1
+
+	b.BatchID = copyBytes(buf[i : i+swarm.HashSize])
+	i += swarm.HashSize
+
+	b.Address = swarm.NewAddress(buf[i : i+swarm.HashSize]).Clone()
+	i += swarm.HashSize
+
+	b.BinID = binary.BigEndian.Uint64(buf[i : i+8])
+
+	return nil
+}
+
+type chunkBinItem struct {
+	Bin     uint8
+	BinID   uint64
+	Address swarm.Address
+}
+
+func (c *chunkBinItem) Namespace() string {
+	return "chunkBin"
+}
+
+// bin/binID
+func (c *chunkBinItem) ID() string {
+	return binIDToString(c.Bin, c.BinID)
+}
+
+func binIDToString(bin uint8, binID uint64) string {
+	binIDBytes := make([]byte, 8)
+	binary.BigEndian.PutUint64(binIDBytes, binID)
+	return path.Join(string(bin), string(binIDBytes))
+}
+
+func (c *chunkBinItem) String() string {
+	return path.Join(c.Namespace(), c.ID())
+}
+
+func (c *chunkBinItem) Clone() storage.Item {
+	if c == nil {
+		return nil
+	}
+	return &chunkBinItem{
+		Bin:     c.Bin,
+		BinID:   c.BinID,
+		Address: c.Address.Clone(),
+	}
+}
+
+const chunkBinItemSize = 1 + 8 + swarm.HashSize
+
+func (c *chunkBinItem) Marshal() ([]byte, error) {
+
+	if c.Address.IsZero() {
+		return nil, errMarshalInvalidAddress
+	}
+
+	buf := make([]byte, chunkBinItemSize)
+	i := 0
+
+	buf[i] = c.Bin
+	i += 1
+
+	binary.BigEndian.PutUint64(buf[i:i+8], c.BinID)
+	i += 8
+
+	copy(buf[i:i+swarm.HashSize], c.Address.Bytes())
+
+	return buf, nil
+}
+
+func (c *chunkBinItem) Unmarshal(buf []byte) error {
+
+	if len(buf) != chunkBinItemSize {
+		return errUnmarshalInvalidSize
+	}
+
+	i := 0
+	c.Bin = buf[i]
+	i += 1
+
+	c.BinID = binary.BigEndian.Uint64(buf[i : i+8])
+	i += 8
+
+	c.Address = swarm.NewAddress(buf[i : i+swarm.HashSize]).Clone()
+
+	return nil
+}
+
+type binItem struct {
+	Bin   uint8
+	BinID uint64
+}
+
+func (b *binItem) Namespace() string {
+	return "binID"
+}
+
+func (b *binItem) ID() string {
+	return string(b.Bin)
```

Comment:
For this.... we need to format the string like we used to in the old code right?
`sync|%03d`

This can lead to incorrect prefix matches. This is for all the binIDs in the keys.

Reply:
i think because the bins are single byte, uint8, this is okay

Reply:
I had to make use of `sync|%03d` because we were using fmt.Sprintf and there was no clear formatting flag for me to use.